### PR TITLE
Use Android friendly libraries. 

### PR DESCRIPTION
**Pull request by syzer**
Added ability to control initial state of picker, without using `ref` , which can be useful when one uses react stateless components (as they do not have `ref`)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -64,6 +62,7 @@ const propTypes = {
   itemStyle: PropTypes.object,
   onSubmit: PropTypes.func,
   disableOverlay: PropTypes.bool,
+	modalVisible: PropTypes.string,
```
Comment (mateioprea) [inline index.js]: Hi @syzer can you remove the space here? let's keep a consistent coding style. Thanks

**Comment by mateioprea** [comment]
Looks fine to me on iOS. Just tested it, but I don't really can test it on Android as I'm having some problems with it. @puredazzle can you help here?

**Comment by puredazzle** [comment]
@mateioprea thanks for testing. @syzer could you review @mateioprea screenshots? I don't know how it should look on an android. :)

**Comment by syzer** [comment]
This is how drop downs look on android... 
We could change it to `spinner` https://developer.android.com/guide/topics/ui/controls/spinner.html, or
we could style that it will look exactly like the IOS one, 
but i just do not see the point.



Btw I just realised that i made error on line 75:
``` js 
modalVisible: PropTypes.modalVisible || false,
```
Should be
```js
modalVisible: this.props.modalVisible || false,
```

I'll make new push to fix it is that ok ?

**Comment by puredazzle** [comment]
@syzer Ok, then we keep it as it is. Yes, please make a push with the fix.

**Comment by syzer** [comment]
@puredazzle ok theres one more commit, 
https://github.com/puredazzle/react-native-simple-picker/pull/24/commits/4eb1af38e9f22ea0c21a7d44585886eeb19574ef

I even managed to squeeze a feature (to overwrite styles) there :)
also fixed `mainBox`, was undefined.

Also as https://facebook.github.io/react-native/docs/picker.html#mode
says, it turns out that android has not only `overlay` mode but also `dropdown`....
we could proxy that prop to picker with spread operator.

@mateioprea Would you be so good and test again if I didn't make error with 
`static defaultProps = {`
JSX should parse it ok.
It `works on my machine` TM :D

**Comment by mateioprea** [comment]
@syzer will try to test it in the next few hours. have some tasks to finish at work first thanks!

**Comment by puredazzle** [comment]
@syzer @mateioprea should there be any radio buttons like in the native picker on Android as seen here 
21895119_10155179548828440_36919392_o
– I have no Android so don't know the expected behaviour.

**Comment by syzer** [comment]
@puredazzle not sure.. in dropdown mode definetly not.
but in overlay mode IDK.
Either way probably next weeks i will send more PR about android

File: index.js [hunk 2]
```diff
@@ -51,33 +53,39 @@ const styles = StyleSheet.create({
...
+  confirmText: PropTypes.string,
+  disableOverlay: PropTypes.bool,
+  initialOptionIndex: PropTypes.number,
   itemStyle: PropTypes.object,
+  labels: PropTypes.array,
+  modalVisible: PropTypes.bool,
   onSubmit: PropTypes.func,
-  disableOverlay: PropTypes.bool,
+	options: PropTypes.array.isRequired,
+	styles: PropTypes.object,
```
Comment (mateioprea) [inline index.js]: Can you please format these lines?

Comment (puredazzle) [reply index.js]: @syzer we have a [editorconfig](https://github.com/puredazzle/react-native-simple-picker/blob/master/.editorconfig) in the project, please try to keep the styling we have. thanks in advance.

File: index.js [hunk 3]
```diff
@@ -51,33 +53,39 @@ const styles = StyleSheet.create({
...
     this.state = {
-      modalVisible: false,
-      selectedOption: this.props.options[selected],
+      modalVisible: props.modalVisible || false,
+      selectedOption: props.options[selected],
     };
+		this.styles = StyleSheet.create({...styles, ...props.styles});
 
-    this.onPressCancel = this.onPressCancel.bind(this);
+		this.onPressCancel = this.onPressCancel.bind(this);
```
[the remark this answers is not in the document]
Comment (syzer) [reply index.js]: also compoanentWillRecieveProps() needs to be updated

**Comment by syzer** [comment]
I fixed and moved all according to editorconfig (with tabs)
added onCancel Handler
I'm using my fork of this library in the meantime.

**Comment by Batuhan-Akkaya** [comment]
Could you share the code of the works on android, please?

**Comment by puredazzle** [comment]
@mateioprea what do you think? should we merge the PR?

**Comment by syzer** [comment]
@puredazzle i guess you guys had different code than editorconfig was configured... :)

**Comment by vinkla** [comment]
@syzer please revert the syntax changes in this pull request. It will make the review process much easier for us to handle.

**Comment by puredazzle** [comment]
@syzer @mateioprea @Batuhan-Akkaya @vinkla I merged the changes and made the style updates to the code. Released a new version 2.0!

**Pull request merged**